models/opensea/ethereum/opensea_v4_ethereum_trades.sql

```diff
@@ -27,526 +25,541 @@ with source_ethereum_transactions as (
+        , case 
+            when sender = receiver then true 
+            else false 
+        end as is_self_trans
+
+        , case 
+            when f.wallet_address is not null then true 
+            else false 
+        end as is_platform_fee
+
+        , case 
+            when item_type in ('native','erc20')
+            then sum(case when item_type in ('native','erc20') then 1 end) over (
+                partition by tx_hash, evt_index, sub_type 
+                order by sub_idx
+                )
+        end as eth_erc_idx
+        
+         , sum(
+             case 
+                when offer_first_item_type = 'erc20' and sub_type = 'consideration' and item_type in ('erc721','erc1155') then 1
+                when offer_first_item_type in ('erc721','erc1155') and sub_type = 'offer' and item_type in ('erc721','erc1155') then 1
+            end) over (
+                partition by tx_hash, evt_index
+        ) as nft_cnt
+        
+        , sum(
+            case 
+                when offer_first_item_type = 'erc20' and sub_type = 'consideration' and item_type in ('erc721') then 1
+                when offer_first_item_type in ('erc721','erc1155') and sub_type = 'offer' and item_type in ('erc721') then 1
+            end) over (
+                partition by tx_hash, evt_index
+        ) as erc721_cnt
+
+        , sum(
+            case 
+                when offer_first_item_type = 'erc20' and sub_type = 'consideration' and item_type in ('erc1155') then 1
+                when offer_first_item_type in ('erc721','erc1155') and sub_type = 'offer' and item_type in ('erc1155') then 1
+            end) over (
+                partition by tx_hash, evt_index
+        ) as erc1155_cnt
+
+        , try_cast(date_trunc('day', a.block_time) as date) as block_date
+
+        , case 
+            when offer_first_item_type = 'erc20' then 'offer accepted'
+            when offer_first_item_type in ('erc721','erc1155') then 'buy'
+            else 'etc' -- some txns has no nfts
+        end as order_type
+
+        , case 
+            when om_order_id = 1 then false
+            when offer_first_item_type = 'erc20' and sub_type = 'offer' and item_type = 'erc20' then true
+            when offer_first_item_type in ('erc721','erc1155') and sub_type = 'consideration' and item_type in ('native','erc20') then true
+            else false
+        end as is_price
+
+        ,case 
+            when om_order_id = 1 then false
+            when offer_first_item_type = 'erc20' and sub_type = 'consideration' and eth_erc_idx = 0 then true  -- offer accepted has no price at all. it has to be calculated.
+            when offer_first_item_type in ('erc721','erc1155') and sub_type = 'consideration' and eth_erc_idx = 1 then true
+            else false
+        end as is_netprice
+
+        , case 
+            when is_platform_fee then false
+            when offer_first_item_type = 'erc20' and sub_type = 'consideration' and eth_erc_idx > 0 then true
```

Comment:
Hi @DevCav,
`eth_erc_idx` at this moment is still at same level computation(at line 271, and this will result in unresolved pointer).
We may need to use nested query  or CTE to calculate `eth_erc_idx` first for the dependency.